lib/gather/commands/show.rb

```diff
@@ -42,7 +42,28 @@ def run
           data = data.deep_merge(Collector.logical_data)
           File.open(Config.data_path, 'w') { |file| file.write(data.to_yaml) }
         end
-        File.open(Config.data_path) { |file| puts file.read }
+
+        data = YAML.load_file(Config.data_path)
+        keys = (@args[0] || '.').delete_prefix('.').split('.')
+
+        value = data
+        keys.each do |key|
+          if key.match(/\[[^\]]*\]/)
+            key = key[1..-2]
+            key = key.to_i if key.to_i.to_s == key
+          else
+            key = key.to_sym
+          end
+          break if value.nil?
+
+          value = value[key]
+        end
+
+        if value.respond_to?(:each)
```

Comment:
Will there only ever be a single level of nesting?

Reply:
I'm not sure what you mean by this - this check is only here to see if the value that `show` is about to give is a hash/array or a singleton value. If it's on its own, it's better to leave it as-is, but if it's an array or hash then we should format it in a more readable way, so it gets converted to YAML to match how `show` normally displays hash data.